actors/sunbird-lms-mw/actors/common/src/main/java/org/sunbird/learner/actors/LearnerStateActor.java

```diff
@@ -343,7 +343,14 @@ private void removeUnwantedProperties(Response response) {
     List<Map<String, Object>> activeCourses =
         (List<Map<String, Object>>) (result.get(JsonKey.CONTENT));
     List<Map<String, Object>> contentsForCourses = getcontentsForCourses(request, activeCourses);
-
+    Set<String> courseIds =
+            contentsForCourses
+                    .stream()
+                    .map(course -> (String) course.get(JsonKey.COURSE_ID))
```

Comment:
there is no "courseId" field in returned courses, hence we have to collect "identifier" field, also make sure contentsForCourses is not null